From bdd526d43fbe76d09816e987999fc5796b9b5e73 Mon Sep 17 00:00:00 2001
From: Ansu Mathew <Ansu_Mathew@comcast.com>
Date: Thu, 20 Mar 2025 07:00:39 +0000
Subject: [PATCH] check session has a valid pointer rebased

---
 Source/ocdm/open_cdm_impl.h | 43 +++++++++++++++++++++++++++----------
 1 file changed, 32 insertions(+), 11 deletions(-)

diff --git a/Source/ocdm/open_cdm_impl.h b/Source/ocdm/open_cdm_impl.h
index 3b997a9..d6517f1 100644
--- a/Source/ocdm/open_cdm_impl.h
+++ b/Source/ocdm/open_cdm_impl.h
@@ -556,13 +556,19 @@ public:
     { 
         ASSERT(_session != nullptr);
 
-        return _session->Metadata();
+        if (_session != nullptr) {
+            return _session->Metadata();
+        }
+        return "";
     }
     inline uint32_t Metricdata(uint32_t& bufferSize, uint8_t buffer[])
     {
         ASSERT(_session != nullptr);
 
-        return _session->Metricdata(bufferSize, buffer);
+        if (_session != nullptr) {
+            return _session->Metricdata(bufferSize, buffer);
+        }
+        return 1;
     }
     inline const string& BufferId() const
     {
@@ -591,39 +597,52 @@ public:
     {
         ASSERT(_session != nullptr);
 
-        _session->Close();
+        if (_session != nullptr) {
+            _session->Close();
+        }
     }
     inline void ResetOutputProtection() {
         ASSERT (_session != nullptr);
 
-        _session->ResetOutputProtection();
+        if (_session != nullptr) {
+            _session->ResetOutputProtection();
+        }
     }
     inline void SetParameter(const std::string& name, const std::string& value)
     {
         ASSERT (_session != nullptr);
 
-        _session->SetParameter(name, value);
+        if (_session != nullptr) {
+            _session->SetParameter(name, value);
+        }
     }
     inline int Remove()
     {
-
         ASSERT(_session != nullptr);
 
-        return (_session->Remove() == 0);
+        if (_session != nullptr) {
+            return (_session->Remove());
+        }
+        return 1;
     }
     inline int Load()
     {
 
         ASSERT(_session != nullptr);
 
-        return (_session->Load() == 0);
+        if (_session != nullptr) {
+            return (_session->Load());
+        }
+        return 1;
     }
     inline void Update(const uint8_t* pbResponse, const uint16_t cbResponse)
     {
 
         ASSERT(_session != nullptr);
 
-        _session->Update(pbResponse, cbResponse);
+        if (_session != nullptr) {
+            _session->Update(pbResponse, cbResponse);
+        }
     }
     uint32_t Decrypt(uint8_t* encryptedData, const uint32_t encryptedDataLength,
         const ::SampleInfo* sampleInfo,
@@ -728,7 +747,9 @@ protected:
 
             ASSERT (_session != nullptr);
 
-            _session->Release();
+            if (_session != nullptr) {
+                 _session->Release();
+            }          
 
             if (_sessionExt != nullptr) {
                 _sessionExt->Release();
@@ -738,7 +759,7 @@ protected:
 
         _session = session;
 
-        if (session != nullptr) {
+        if (session != nullptr && _session != nullptr) {
 
             _session->AddRef();
             _sessionExt = _session->QueryInterface<Exchange::ISessionExt>();
-- 
2.48.0

