Index: ocdm-nova/CMakeLists.txt
===================================================================
--- ocdm-nova.orig/CMakeLists.txt
+++ ocdm-nova/CMakeLists.txt
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.10.3)
-project(ocdm-NoVA)
+project(ocdm-NoVA VERSION 0.1.0)
+set(ocdm-NoVA_SOVERSION 0)
 
 set(CMAKE_CXX_STANDARD 14)
 enable_language(CXX)
@@ -20,14 +21,18 @@ set(LIB_OCDM_NOVA_SOURCES
         source/open_cdm.cpp
         source/open_cdm_adapter.cpp
         source/open_cdm_ext.cpp
-
         source/OpenCDMSession.cpp
-        ${LIBNOVA_HEADERS})
+        ${LIBNOVA_HEADERS}
+        )
 
 set(LIB_OCDM_NOVA_PUBLIC_HEADERS
+        include/opencdm/open_cdm.h
+        include/opencdm/open_cdm_adapter.h    
         )
 
-add_library(ocdm-NoVA SHARED ${LIB_OCDM_NOVA_SOURCES} )
+add_library(ocdm-NoVA SHARED ${LIB_OCDM_NOVA_SOURCES})
+
+set_target_properties(ocdm-NoVA PROPERTIES VERSION ${PROJECT_VERSION})
 
 install(TARGETS ocdm-NoVA LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR})
@@ -35,8 +40,13 @@ install(TARGETS ocdm-NoVA LIBRARY
 target_link_libraries(ocdm-NoVA PUBLIC
        NoVA)
 
-InstallPackageConfig(
-        TARGETS ocdm-NoVA
-        DESCRIPTION "OCDM library")
+configure_file(
+       "pkg-config/ocdm-NoVA.pc.in"
+       "pkg-config/ocdm-NoVA.pc"
+       @ONLY)
+
+install(FILES ${LIB_OCDM_NOVA_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/opencdm)
 
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pkg-config/ocdm-NoVA.pc"
+       DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
Index: ocdm-nova/pkg-config/ocdm-NoVA.pc.in
===================================================================
--- /dev/null
+++ ocdm-nova/pkg-config/ocdm-NoVA.pc.in
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name:  ocdm-NoVA
+Description: OCDM implementation based on NoVA library
+Version: @JSONCPP_VERSION@
+Libs: -L${libdir} -locdm-NoVA
+Cflags: -I${includedir}
+
